payload/cache/payload_id.go

```diff
@@ -83,6 +84,9 @@ func (p *PayloadIDCache[RootT, SlotT]) Get(
 	if !ok {
 		return engineprimitives.PayloadID{}, false
 	}
+
+	// Successfully retrieved. Remove from cache.
+	delete(p.slotToStateRootToPayloadID[slot], stateRoot)
```

Comment:
another nit, possibly unrelated: go should allow using structs as keys, so should we turn this map of maps into a map of `struct{slot, root}`?

Reply:
we could keep the struct only internal and keep the cache API the same (passing slot and root separately is fine)